src/Cache/Engine/RedisEngine.php

```diff
@@ -228,14 +228,23 @@ public function clear($check)
         if ($check) {
             return true;
         }
-        $keys = $this->_Redis->getKeys($this->_config['prefix'] . '*');
 
-        $result = [];
-        foreach ($keys as $key) {
-            $result[] = $this->_Redis->del($key) > 0;
-        }
+        $isAllDeleted = true;
+        $iterator = null;
+        $pattern = $this->_config['prefix'] . '*';
+        do {
+            $keys = $this->_Redis->scan($iterator, $pattern);
+
+            // Redis may return empty results, so protect against that
+            if ($keys !== false) {
```

Comment:
Is there a scenario where `$keys == false` and `$iterator > 0`? It looks like you have a potential infinite loop. You could do something like

```php
if ($keys === false || $iterator === 0) {
  break;
}
```

to avoid this scenario.

Reply:
According to the PHPRedis [documentation](https://github.com/phpredis/phpredis#example-6) this should not be possible.
`$iterator` loops over _all_ keys, while `$keys` only contain results that matches the pattern, within the current iterator window.

I could simplify this method by using the `Redis::SCAN_RETRY` option if you like, because currently it does indeed look a bit strange.

Reply:
Using SCAN_RETRY sounds like a good idea to me.